authentik/api/renderers.py

```diff
@@ -0,0 +1,64 @@
+import contextlib
+from datetime import timedelta
+from decimal import Decimal
+from typing import Any
+
+import orjson
+from django.db.models.query import QuerySet
+from django.utils.encoding import force_str
+from django.utils.functional import Promise
+from rest_framework.compat import coreapi
+from rest_framework.renderers import JSONRenderer as BaseJSONRenderer
+
+
+def default(obj: Any) -> Any:
+    """
+    Render object to JSON.
+
+    Adaptation of https://github.com/encode/django-rest-framework/blob/master/rest_framework/utils/encoders.py
```

Comment:
Instead of having all the cases in here, couldn't we just call the upstream function except for types that it isn't needed for?

Reply:
We could, the main issue is that the upstream one returns `super().default(obj)` which we may not want.

Reply:
No, actually we could

Reply:
Yeah, I tried, I can't get it to work with generators :/

Reply:
maybe we can have a look at (or even just use) https://github.com/brianjbuck/drf_orjson_renderer, since they don't seem to need this (also that repo switches the request parser to orjson too which we might as well do too)